### ISSUE TITLE
client_id is needed when using an authorization_code

### ISSUE THREAD

**Issue by bbalet** [bshaffer/oauth2-server-php-docs#96]
I don't know if it is a bug or a mistake into Cookbook (Step-By-Step Walkthrough), but when you call this one:

    curl -u testclient:testpass http://localhost/token.php -d 'grant_type=authorization_code&code=YOUR_CODE'

It fails. I had to repeat the client_id and it worked.

**Comment by alpha1125** [comment]
I could not reproduce OP's issue.

1. Ran authorized.php in browser gave me back a code
2. used code in
```
curl -u testclient:testpass http://walkthrough.local/token.php -d 'grant_type=authorization_code&code=[CODE]'
```
3. gave me back a token.

**Comment by bbalet** [comment]
How many clients into your db?

**Comment by alpha1125** [comment]
only 1... I was testing it out, haven't used in in more than one scenario now... I'll test out more clients.

**Comment by bluebaroncanada** [comment]
Is this still an issue?

**Comment by bshaffer** [comment]
Yes I'm very curious if this was resolved. It could be the apache headers issue (documented [here](https://github.com/bshaffer/oauth2-server-php/blob/develop/src/OAuth2/Request.php#L170)).

**Comment by daniel-exsyn** [comment]
Could it be that you have the ``OAuth2\OpenID\GrantType\AuthorizationCode`` grant type instead of ``OAuth2\GrantType\AuthorizationCode``?
